filebeat/input/filestream/internal/input-logfile/harvester.go

```diff
@@ -204,7 +205,7 @@ func startHarvester(
 			if errors.Is(err, ErrHarvesterAlreadyRunning) {
 				return nil
 			}
-
+			ctx.UpdateStatus(status.Degraded, err.Error())
 			return fmt.Errorf("error while adding new reader to the bookkeeper %w", err)
```

Comment:
By just reporting the message from `err.Error()` we're not sending enough context for the user, it is best to also add the message from the returned error:
```suggestion
			err = fmt.Errorf("error while adding new reader to the bookkeeper %w", err)
			ctx.UpdateStatus(status.Degraded, err.Error())
			return err
```

The same applies to all other places you added a `ctx.UpdateStatus` right above an error return.

Reply:
Done.